pkg/resource/container_image.go

```diff
@@ -0,0 +1,426 @@
+		return err
+	}
+
+	return nil
+}
+
+// The decompression process is a little more involved so separated out.
+func (r *ContainerImage) decompress(t io.Reader, layer string, mediaType string, size int64) error {
+	// The maximum file size should be less than 10x the layer size.
+	size = size * 10
+	path := filepath.Join(r.ClonePath(), layer)
+
+	var tarReader *tar.Reader
+
+	if strings.Contains(strings.ToLower(mediaType), "gzip") {
+		gzReader, err := gzip.NewReader(t)
+		if err != nil {
+			return err
+		}
+		tarReader = tar.NewReader(gzReader)
+		defer gzReader.Close()
+	} else {
+		tarReader = tar.NewReader(t)
+	}
+
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return err
+		}
+		path, err := sanitizePath(path, header.Name)
+		if err != nil {
+			logger.Error("%v - skipped", err)
+			continue
+		}
+		info := header.FileInfo()
+		if info.IsDir() {
+			if err = os.MkdirAll(path, 0700); err != nil {
+				return err
+			}
+			continue
+		}
+
+		file, err := os.OpenFile(path, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0600) // #nosec G304
+		if err != nil {
+			logger.Error("%v", err)
+			continue
+		}
+		defer file.Close()
+
+		// Copy a maximum number of bytes (layer size * 10) so we do not get "bombs". It is unlikely that a file
+		// with significant entropy will be compressed more than 10x. We can review this.
+		n, err := io.CopyN(file, tarReader, size)
+		if err != nil && err != io.EOF {
+			return fmt.Errorf("could not copy file to disk: %v", err)
+		}
+		if n >= size {
+			logger.Warning("copying file %s did not finish due to max file size: %v", file.Name(), err)
+		}
+	}
+	return nil
+}
+
+// ClonePath returns where this repo has been cloned if cloned else ""
+func (r *ContainerImage) ClonePath() string {
+	return r.clonePath
+}
+
+// Depth returns the depth for things that have version control
+func (r *ContainerImage) Depth() uint16 {
+	return 0
+}
+
+// EnrichResult adds contextual information to each result
+func (r *ContainerImage) EnrichResult(result *response.Result) *response.Result {
+
+	hash, file, found := strings.Cut(result.Location.Path, string(os.PathSeparator))
+	if found {
+		result.Location.Version = hash
+		result.Location.Path = file
+	}
+	result.Notes = r.labels
+	if result.Location.Version != "" {
+		// If there is no layer then it is a metadata file
+		result.Kind = response.ContainerMetdataResultKind
+	}
+	result.Kind = response.ContainerLayerResultKind
+
+	result.Contact.Name, result.Contact.Email = r.Contact()
+
+	return result
+}
+
+// SetDepth allows you to adjust the depth for the resource
+func (r *ContainerImage) SetDepth(depth uint16) {
+	// no-op
+}
+
+// SetCloneTimeout lets you adjust the timeout before the clone aborts
+func (r *ContainerImage) SetCloneTimeout(timeout time.Duration) {
+	// no-op
+}
+
+// Since returns the date after which things should be scanned for things
+// that have versions
+func (r *ContainerImage) Since() string {
```

Comment:
I take it there's no timestamp metadata about the layers to say, I only want to scan layers built since yesterday or something like that, correct?

Reply:
No, just a type, digest and size.